portal-impl/test/unit/com/liferay/portal/util/PortalImplEscapeRedirectTest.java

```diff
@@ -59,13 +64,22 @@ public void setUp() {
 		_originalRedirectURLSettings = ReflectionTestUtil.getAndSetFieldValue(
 			RedirectURLSettingsUtil.class, "_redirectURLSettings",
 			_redirectURLSettingsImpl);
+
```

Comment:
Use mock here because when we call _portalImpl.escapeRedirect()

**before upgrading virtual.hosts.vali.hosts**
in `if (!_validPortalDomainCheckDisabled && isValidPortalDomain(domain)) `, `_validPortalDomainCheckDisabled` is set to be true, and this condition will be false, and just keep moving to next line of code.

**after upgrading virtual.hosts.vali.hosts**
in `if (!_validPortalDomainCheckDisabled && isValidPortalDomain(domain)) `, `_validPortalDomainCheckDisabled` is set to be false, and it will go into isValidPortalDomain(domain), and in this method, we will throw NPE. We need to use mock to get rid of NPE, and make sure it can keep moving to next line of code as before